qiskit/circuit/parametertable.py

```diff
@@ -79,3 +84,208 @@ def __len__(self):
 
     def __repr__(self):
         return 'ParameterTable({})'.format(repr(self._table))
+
+
+def _deprecated_set_method():
```

Comment:
Is this different to `qiskit.utils.deprecation.deprecate_function`?

Reply:
It allows to print the method name in without having to hardcode it in the deprecation call. E.g. a solution using `deprecate_function` could be
```python
def deprecation_msg(funcname):
    return f'The ParameterView.{funcname} method is deprecated as of '
            'Qiskit Terra 0.17.0 and will be removed no sooner than 3 months '
            'after the release date. Circuit parameters are returned as View '
            'object, not set. To use set methods you can explicitly cast to a '
            'set.'

# inside ParameterView
@deprecate_function(deprecation_msg('ParameterView.difference'))
def difference(...)
```
I can change it to that if you prefer, I was just being lazy and added a quick new method instead of having to type out the name of every method I want to deprecated 😉